lib/ide/json-rpc/src/handler.rs

```diff
@@ -150,55 +128,135 @@ pub type OngoingCalls = HashMap<Id,oneshot::Sender<ReplyMessage>>;
 /// `Notification` is a type for notifications. It should implement
 /// `DeserializeOwned` and deserialize from JSON maps with `method` and `params`
 /// fields.
+///
 #[derive(Debug)]
 pub struct Handler<Notification> {
-    /// Contains handles to calls that were made but no response has came.
-    pub ongoing_calls   : OngoingCalls,
-    /// Provides identifiers for requests.
-    pub id_generator    : IdGenerator,
-    /// Transports text messages between this handler and the peer.
-    pub transport       : Box<dyn Transport>,
-    /// Allows receiving events from the `Transport`.
-    pub incoming_events : std::sync::mpsc::Receiver<TransportEvent>,
-    /// Handle to send outgoing events.
-    pub outgoing_events : Option<UnboundedSender<Event<Notification>>>,
+    rc: Rc<RefCell<HandlerData<Notification>>>,
+}
+
+impl<Notification> Handler<Notification> {
```

Comment:
We've got two impls block - please merge them or add subsections `// ===` between them